dts/bindings/misc/nxp,inputmux.yaml

```diff
@@ -0,0 +1,43 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+include: [base.yaml]
+
+compatible: "nxp,inputmux"
+description: NXP i.MX INPUTMUX driver. Routes internal SoC signals.
+
+properties:
+  reg:
+    required: true
+
+child-binding:
+  description: |
+    INPUTMUX attachment or signal enablement.
+
+    For signal enablement, specify the "signal" property and use "MUX_*_ENA"
+    constants defined in used device's DT binding for this driver. For example:
+
+    flexcomm12_dma0_rx {
+        signal = <MUX_FLEXCOMM12_RX_TO_DMAC0_CH34_REQUEST_ENA>;
+    };
+
+    For signal attachment (routing), specify the "selector" and "source"
+    properties. The "selector" property specifies which signal is to be
+    assigned and the "source" property specifies the source to which the
+    assignment will be made. For example:
+
+    dsp_int0 {
+       selector = <0>;
+       source = <MUX_FLEXCOMM0_TO_DSP_INTERRUPT>;
+    };
```

Comment:
the input mux node should have one property which is simply an array of pairs of registers and values, no child nodes

Reply:
IE we should simply configure all inputmux setting in one property?

I considered this as well, but I think the separate nodes makes sense- if you have one property for all the settings of the inputmux, than if you'd like to add or change one setting you need to duplicate the entire list of inputmux settings for your system. This seems more error prone than having inputmux settings be in separate nodes, because that way they can be additive (IE you only need to include the setting you want to add in an overlay)

Reply:
yes, it would be like there is an input mux node, with children that represent the muxes, and they have states that represent which thing to mux to, and the other nodes reference their mux with what state they want to set to, something like this